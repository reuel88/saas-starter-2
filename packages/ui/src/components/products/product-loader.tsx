import { FC } from "react";
import { Card, CardFooter, CardHeader } from "../card";
import { Skeleton } from "../ui/skeleton";

export type ProductLoaderProps = {};

export const ProductLoader: FC<ProductLoaderProps> = () => {
  return (
    <Card>
      {/* Image */}
      <Skeleton className="aspect-square" />

      <CardHeader className="space-y-0">
        {/* Title */}
        <div className="py-1">
          <Skeleton className="h-4" />
        </div>

        {/* Price & Rating */}
        <div className="py-[5px]">
          <Skeleton className=" h-[18px]" />
        </div>
      </CardHeader>

      {/* CTA */}
      <CardFooter>
        <Skeleton className="h-10 w-full" />
      </CardFooter>
    </Card>
  );
};
