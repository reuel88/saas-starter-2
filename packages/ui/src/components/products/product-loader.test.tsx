import { render } from "@testing-library/react";
import { ProductLoader } from "./product-loader";

describe("ProductLoader", () => {
  it("should render correctly", () => {
    // Arrange

    // Act
    const { baseElement } = render(<ProductLoader />);

    // Assert
    expect(baseElement).toMatchSnapshot();
  });
});
