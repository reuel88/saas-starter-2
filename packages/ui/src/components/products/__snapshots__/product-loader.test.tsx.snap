// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductLoader > should render correctly 1`] = `
<body>
  <div>
    <div
      class="rounded-lg border bg-card text-card-foreground shadow-sm"
    >
      <div
        class="animate-pulse rounded-md bg-muted aspect-square"
      />
      <div
        class="flex flex-col p-6 space-y-0"
      >
        <div
          class="py-1"
        >
          <div
            class="animate-pulse rounded-md bg-muted h-4"
          />
        </div>
        <div
          class="py-[5px]"
        >
          <div
            class="animate-pulse rounded-md bg-muted h-[18px]"
          />
        </div>
      </div>
      <div
        class="flex items-center p-6 pt-0"
      >
        <div
          class="animate-pulse rounded-md bg-muted h-10 w-full"
        />
      </div>
    </div>
  </div>
</body>
`;
